### DISCUSSION TITLE
collection scan

### DISCUSSION THREAD

**Discussion by vadicamo** [lucaro/DRES#451] [General]
Hi all,

in  last version of DRES   I  struggled a lot with the command 
“DRES> collection scan -c NameCollection -vt mp4” 
without success (“command not found”). 

In the end, I used  
“DRES> collection scan -i idCollection -vt mp4” 
which worked 

Just wondering if this is my problem or a general one, or if there is a different command to be used to scan a collection using the collection name and not the ID.

Thanks,
Lucia

**Comment by lucaro** [comment]
Dear Lucia,

interesting, what was the name of the collection? Also, what version are you using, 1.3 or 2.0RC1?

**Comment by vadicamo** [comment]
Hi , I used  v2.0.0-RC1 (the last commit f6ef379c51b63a69d7146d3ce811062ba53dbf67)

This was my first time downloading and using DRES, so I executed the following steps.

1. The original  [config.json](https://github.com/dres-dev/DRES/blob/master/config.json) was not working for me so I modified it into
`{
    "dataPath" : "/dres-data/data",
    "cache": {
        "cachePath" : "/dres-data/cache"
    },
    "eventsLocation": "/dres-data/events",
    "auditLocation": "/dres-data/audit",
    "externalMediaLocation": "/dres-data/external",
    "enableSsl" : false
 }`
2. I created an admin user `user create -u admin -p mypsw -r ADMIN`
4. I created a participant user `user create -u user1 -p mypsw -r PARTICIPANT`
5. I created a collection named mvk:  `collection create -n mvk -d "Marine Video Kit Collection used in VBS"    -p /my_video_path `
6.  `collection scan -c mvk -vt mp4` 
output: _invalid command: null_
7.  `collection scan -h`  
 Output:
 _Scans a collection directory and adds found items
Options:
  -i, --id=<text>           ID of a media collection.
  -it, --imageType=<value>  Image file types (endings) to be considered in the scan
  -vt, --videoType=<value>  Video file types (endings) to be considered in the scan
  -h, --help                Show this message and exit_

Since the '-c' option was not presente in the help I used `collection list` to get the ID and then scan the data with 
` collection scan -i myCollectionIDnumber -vt mp4`
7. I was able to scan the data but got errors in using ffmpeg, so I also modified the [Dockerfile](https://github.com/dres-dev/DRES/blob/master/Dockerfile) by adding  ` RUN chmod +x ./dres-dist/lib/ffmpeg/ff*` 
8. after rebuilding ` collection scan -i myCollectionIDnumber -vt mp4` worked  well

**Reply by lucaro** [reply]
Yes, the -c option does not exist anymore. We streamlined some parameters.
The Docker file is also out of date, we still need to update that. Then again, I would recommend using it outside of a container anyway since it makes interaction with collections easier. For internal use, there really isn't any benefit of containerizing it, the only system dependency it has is java.
